docs/docs/grpc.md

```diff
@@ -18,12 +18,134 @@ main = runGRpcApp 8080 "helloworld" quickstartServer
 
 ## Building a client
 
--- TODO:
+Right now there are two options for building clients: using records or with `XTypeApplications`. To give a proper introduction to both options let's consider in detail an example client for the following services:
+
+```protobuf
+service Service {
+  rpc getPerson (PersonRequest) returns (Person);
+  rpc newPerson (Person) returns (PersonRequest);
+  rpc allPeople (google.protobuf.Empty) returns (stream Person);
+}
+```
+
+Regardless of the approach we decide to use, we can construct a basic CLI for the client this way:
+
+```haskell
+import System.Environment
+
+main :: IO ()
+main = do
+  let config = grpcClientConfigSimple "127.0.0.1" 8080 False
+  Right client <- setupGrpcClient' config
+  args <- getArgs
+  case args of
+    ["watch"]       -> watching client
+    ["get", idp]    -> get client idp
+    ["add", nm, ag] -> add client nm ag
+    _               -> putStrLn "unknown command"
+```
+
+Where `watch`, `get` and `add` are the only valid 3 commands that our CLI is going to accept and call each respective service.
 
 ### Using records
 
--- TODO:
+This option is a bit more verbose but it's also more explicit with the types and _"less magic"_ than the one with `TypeApplications`.
+
+We need to define a new record type (hence the name) that declares the services our client is going to consume:
+
+```haskell
+import GHC.Generics (Generic)
+import Mu.GRpc.Client.Record
+
+data Call = Call
+  { getPerson :: MPersonRequest -> IO (GRpcReply MPerson)
+  , newPerson :: MPerson -> IO (GRpcReply MPersonRequest)
+  , allPeople :: IO (ConduitT () (GRpcReply MPerson) IO ())
+  } deriving Generic
+```
+
+Note that we had to derive `Generic`. We also need to tweak a little bit our `main` function:
+
+```diff
+main :: IO ()
+main = do
+   let config = grpcClientConfigSimple "127.0.0.1" 1234 False
+-  Right client <- setupGrpcClient' config
++  Right grpcClient <- setupGrpcClient' config
++  let client = buildService @Service @"" grpcClient
+   args <- getArgs
+```
+
+Instead of building our client directly, we need to use `TypeApplications` (yes, we can't escape from `TypeApplications` either way 🤷🏼‍♂️) in a call to `buildService` to create the actual gRPC client.
+
+After that, the implementation of the three service calls becomes quite trivial:
+
+```haskell
+import Text.Read (readMaybe)
+
+get :: Call -> String -> IO ()
+get client idPerson = do
+  let req = MPersonRequest $ readMaybe idPerson
+  putStrLn $ "GET: is there some person with id: " ++ idPerson ++ "?"
+  res <- getPerson client req
+  putStrLn $ "GET: response was: " ++ show res
+```
 
-### Using `TypeApplications`
+Notice the use of `readMaybe` to cast the strings to the appropiate type in a safe manner! 👆🏼
```

Comment:
```suggestion
Notice the use of `readMaybe` to convert the strings to the appropiate type in a safe manner! 👆🏼
```

Just to avoid people thinking we're `unsafeCoerce` ing here.

Reply:
Thanks pepe! ❤️